src/client/components/App.tsx

```diff
@@ -84,6 +90,10 @@ class App extends React.PureComponent<AppProps, AppState> {
         this.setState({ showSettingsPane: !this.state.showSettingsPane });
     };
 
+    addBreakpoint = (b: number): void => {
+        this.state.breakpoints.push(b);
+    };
+
```

Comment:
Mutating state variables in React is a little dangerous; technically the state has changed, but because setState was not called, React doesn't realize anything has changed and a re-render is not triggered. You should create a new array instead, like so:

```
addBreakpoint = (b: number): void => {
    this.setState({ breakpoints: [ ...this.state.breakpoints, b ] });
};
```